engine/src/main/java/com/datatorrent/stram/debug/TupleRecorder.java

```diff
@@ -463,10 +492,21 @@ private void publishTupleData(int portId, Object obj)
         wsClient.publish(recordingNameTopic, map);
       }
     } catch (Exception ex) {
-      logger.warn("Error publishing tuple data", ex);
+      if (shouldLog()) {
+        logger.warn("Error publishing tuple data {}", obj, ex);
+      }
     }
   }
 
+  private boolean shouldLog()
+  {
+    if ((lastLog == -1) || (totalTupleCount - lastLog) >= ERROR_LOG_GAP) {
```

Comment:
Correct, after every logged error, it will be necessary to decrement the gap number of tuples. It is minimal overhead compared to recording the same number of tuples. My concern is the dependency on totalTupleCount.